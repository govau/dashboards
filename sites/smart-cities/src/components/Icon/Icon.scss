@import '~style/main';

.svg {
  display: inline-block;
  vertical-align: middle;

  * {
    fill: currentColor;
  }
}

.indicatorMark {
  path { // the "C" or "P"
    fill: $color-white;
  }
}

.arrowInCircle {
  path {
    fill: none;
    stroke: $color-white;
  }
}

.arrowInCircleInverted {
  circle {
    fill: $color-white;
  }

  path {
    fill: none;
    stroke: currentColor;
  }
}

.sunSparkles {
  transform: rotate(0);
  transform-box: fill-box;
  transform-origin: center;
  transition: transform 500ms;

  svg:hover & {
    transform: rotate(90deg);
  }
}

// This will be applied to the SVG element itself
:global(.circleWithShadow) {
  @include shadow(1);

  border-radius: 50%;
  transition: box-shadow $animation-duration;

  &:hover {
    @include shadow(2);
  }
}
