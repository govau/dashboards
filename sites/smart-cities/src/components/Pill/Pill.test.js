import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Pill';

const defaultProps = {
  icon: 'questionMark',
};

it('should match Snapshot', () => {
  const renderedValue =  renderer.create(<Icon {...defaultProps} />).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
