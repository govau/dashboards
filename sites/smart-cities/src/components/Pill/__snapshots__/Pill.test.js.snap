// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<span
  className="wrapper"
  style={
    Object {
      "borderRadius": 15,
      "lineHeight": "26px",
      "minHeight": 30,
    }
  }
/>
`;
